### PR TITLE
Validate and fix up tuples with non-nullable elements

### PR DESCRIPTION
**Pull request by tlively**
The code validating and fixing up non-nullable locals previously did not
correctly handle tuples that contained non-nullable elements, which could have
resulted in invalid modules going undetected. Update the code to handle tuples
and add tests.

### CODE REVIEW

**Comment by tlively** [comment]
Current dependencies on/for this PR:
* main
  * **PR #5909** <a href="https://app.graphite.dev/github/pr/WebAssembly/binaryen/5909" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/WebAssembly/binaryen/5909?utm_source=stack-comment).

File: src/ir/type-updating.cpp [hunk 1]
```diff
@@ -385,21 +387,48 @@ void handleNonDefaultableLocals(Function* func, Module& wasm) {
...
     return Builder(wasm).makeRefAs(RefAsNonNull, get);
   }
+  if (get->type.isTuple()) {
+    auto type = get->type;
+    get->type = getValidLocalType(type, wasm.features);
+    std::vector<Expression*> elems(type.size());
+    Builder builder(wasm);
+    for (Index i = 0, size = type.size(); i < size; ++i) {
+      auto* elemGet =
+        i == 0 ? get : builder.makeLocalGet(get->index, get->type);
```
Comment (kripken) [inline src/ir/type-updating.cpp:424]: I'm confused.  If the tuple has 3 elements, doesn't this look treat 0 as the (fixed up) tuple, and then 1 and 2 as the 2nd and 3rd indexes? What about the 1st index in the tuple?

That is, the first item in this loop has tuple type while all others do not..?

Comment (tlively) [reply src/ir/type-updating.cpp:424]: The original `get` and all the new `get`s have the same full tuple type. It's the tuple extract below that picks a single element out of it.

Comment (kripken) [reply src/ir/type-updating.cpp:424]: Oh, right... sorry, I was very confused here.

**Review by kripken — APPROVED** [review]

**Comment by tlively** [comment]
The fuzzer did find a bug where I never updated the code that updates tees. I have a fix, but producing a minimized test case is taking a while.

**Comment by tlively** [comment]
Going ahead and landing this. The new code is extremely similar to existing code and it is well-tested. The fuzzer seems happy with it so far.

**Pull request merged**